Update C++ rules (#3190)
SonarSource/rspec: rules/S3608/cfamily/rule.adoc

== Why is this an issue?

Lambdas can use variables from their enclosing scope (called "capture") either by reference or by value. Since lambdas may run asynchronously, reference capture should be used with caution because, by the time the lambda runs, the referenced variable may be out of scope, resulting in an access violation at run time.

> You can specify default capture by reference (``++[&]++``), or by value (``++[=]++``). Clearly, default reference capture can cause scope issues, but default value capture can also lead to problems. That's because both forms of default capture implicitly also capture ``++*this++``, which would automatically be used if, for example, you referenced a method from the enclosing scope.

Reviewer: ```suggestion
You can specify default capture by reference (``++[&]++``), or by value (``++[=]++``). Default reference capture can cause scope issues, but so can default value capture. Both forms of default capture implicitly also capture ``++*this++``, which would automatically be used if, for example, you referenced a method from the enclosing scope.
```

Author: Better, thanks!
